middle_end/flambda2/types/provers.ml

```diff
@@ -157,28 +157,55 @@ let prove_get_tag env t = as_property (prove_get_tag_generic env t)
 
 let prove_naked_immediates_generic env t : Targetint_31_63.Set.t generic_proof =
   match expand_head env t with
-  | Naked_immediate (Ok (Naked_immediates is)) ->
+  | Naked_immediate (Ok { immediates = Known is; relations = _ }) ->
+    (* No attempt at reduction *)
```

Comment:
This might be a bit too fidly not to reduce in that case. It would be simpler to just always reduce